subsys/bluetooth/host/Kconfig

```diff
@@ -320,6 +320,16 @@ config BT_SIGNING
 	  This option enables data signing which is used for transferring
 	  authenticated data in an unencrypted connection.
 
+config BT_SMP_PAIRING_ALLOW
+	bool "Allow pairing to proceed"
```

Comment:
Do you think the term "validate" might be more intuitive than "allow" here? Since essentially the callback is validating the pairing parameters against what it considers acceptable. I'd also add "param" or "req" somewhere in there (also in the code) to make it clear that it's validating the parameters of the pairing request rather than the pairing itself.

Reply:
We did have some internal talks about naming and changed to allow, as it is free for the application to include these during processing or not. Also, it is planned to extend this to include security_req also (with AuthReq only), keeping the same flag. So, yes, naming was considered.

Reply:
The reason "allow" bothers me is that it describes one of two possible results of the action, not the action itself. The action is something like "verify", "confirm" or "validate", and the result can be either "allow" or "deny". Calling the option PAIRING_DENY would be equally valid using the current reasoning, which I hope you agree would look strange. So I'd rather have this refer to the action and not the result.

Note: the reason I'm being slightly pedantic about this is that this becomes a public API, and those are always painful to change/correct after the fact. I'd rather we get this right from the beginning.

Reply:
Well, "confirm" is already taken (although biased towards "allow", as well are "verify" and "validate") and I wanted to avoid confusion with it.
So, lifting this to a holistic synonym of taking a decision we also have "resolve", "resolution", "conclude", "regulate", "react", "consent", "authorize", "authenticate" and probably others too.

Reply:
Would prefixing this with "pre" make it any better? E.g. pre_confirm?

Reply:
Maybe we could include where in the pairing stage this is to make it more clear?
pairing_features_confirm, or some other word for confirm? pairing_feature_exchange_confirm?
Maybe accept could work in place of confirm?

Reply:
Well, "confirm" in its entirety does not adhere to the argument of an action based and unbiased naming convention.
Maybe a combination as @joerchan suggests, such as "pairing_properties_audit", "pairing_features_filter" or the like - maybe with "peer" inserted, if not redundant.

Reply:
@jhedberg let's resolve this naming issue with a name that suits you best.